src/main/java/ch/uzh/ifi/hase/soprafs24/service/TileService.java

```diff
@@ -0,0 +1,52 @@
+package ch.uzh.ifi.hase.soprafs24.service;
+
+import ch.uzh.ifi.hase.soprafs24.entity.Tile;
+import ch.uzh.ifi.hase.soprafs24.repository.TileRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@Transactional
+public class TileService {
+
+    private final TileRepository tileRepository;
+
+    @Autowired
+    public TileService(@Qualifier("tileRepository") TileRepository tileRepository) {
+        this.tileRepository = tileRepository;
+    }
+
+    public void drawTile(String sessionID, Integer type) {
+        Tile newTile = new Tile();
+        newTile.setTileID(UUID.randomUUID().toString());
+        newTile.setSessionID(sessionID);
+        newTile.setType(type);
+        newTile.setIsPlaced(false);
+
+        tileRepository.save(newTile);
+        tileRepository.flush();
+    }
```

Comment:
In this implementation, drawTile assumes there is infinite amount of tile to be drawn right? In this case, the game would go on forever because every player can draw as many tile as he wants, we will never reach the end scenario "no hand tiles to place" or "no tile to draw". Is this a new design decision?

Reply:
We talked about the decision in person where all of the logic happens on the client side. This includes the client checking when drawing a tile which tiles he is allowed to ask for/which tiles are not yet set or in another players hand.

Reply:
Know is easy to forget or lose sight of the big picture ideas. I also had the same thought and implemented it wrong but @C0DECYCLE reminded me of the fact, that the client does the logic and the server is just a database.

Hope this helps :)

Reply:
Great got it!